src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs

```diff
@@ -1,17 +1,52 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations.Schema;
+using System.Net.Mail;
 
 namespace SFA.DAS.ApprenticeCommitments.Data.Models
 {
-    [Table("Apprentice")]
     public class Apprentice
     {
-        public long Id { get; set; }
-        public string FirstName { get; set; }
-        public string LastName { get; set; }
-        public Guid UserIdentityId { get; set; }
-        public string Email { get; set; }
-        public DateTime DateOfBirth { get; set; }
-        public DateTime CreatedOn { get; private set; }
+        private Apprentice()
+        {
+        }
+
+        public Apprentice(string firstName, string lastName, Guid userIdentityId, MailAddress email, DateTime dateOfBirth)
+        {
+            FirstName = firstName;
+            LastName = lastName;
+            UserIdentityId = userIdentityId;
+            Email = email;
+            DateOfBirth = dateOfBirth;
+            PreviousEmails = new[] { new ApprenticeEmailHistory(email) };
+        }
+
+        public long Id { get; private set; }
+        public string FirstName { get; private set; }
+        public string LastName { get; private set; }
+        public Guid UserIdentityId { get; private set; }
+        public MailAddress Email { get; private set; }
+        public ICollection<ApprenticeEmailHistory> PreviousEmails { get; private set; }
+        public DateTime DateOfBirth { get; private set; }
+        public DateTime CreatedOn { get; private set; } = DateTime.UtcNow;
+
+        internal void UpdateEmail(MailAddress email)
+        {
+            Email = email;
+            PreviousEmails.Add(new ApprenticeEmailHistory(Email));
+        }
```

Comment:
Just a thought, if the new email address is the same as the existing email address do we want to add a record to the history table? This might happen if a polly retry occurs or we consume the change via a message. Would it be appropriate for it to simply ignore the action. 

Also now we are starting to update records we might want to talk about concurrency but it's not a big issue in this system.

Reply:
Good idea, I added a check